src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -54,11 +56,44 @@ public DocumentPutAction(DocumentsStorage documentsStorage, DocumentDatabase doc
             }
         }
 
+        private struct CompareClusterTransactionId
+        {
+            private readonly ServerStore _serverStore;
+            private readonly DocumentPutAction _parent;
+
+            public CompareClusterTransactionId(DocumentPutAction parent)
+            {
+                _serverStore = parent._documentDatabase.ServerStore;
+                _parent = parent;
+            }
+
+            public void ValidateAtomicGuard(string id, NonPersistentDocumentFlags nonPersistentDocumentFlags, string changeVector)
+            {
+                if (nonPersistentDocumentFlags != NonPersistentDocumentFlags.None) // replication or engine running an operation, we can skip checking it 
+                    return;
+
+                long indexFromOChangeVector = ChangeVectorUtils.GetEtagById(changeVector, _parent._documentDatabase.ClusterTransactionId);
+                if (indexFromOChangeVector == 0)
+                    return;
+                
+                using var _ = _serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext clusterContext);
+                clusterContext.OpenReadTransaction();
```

Comment:
Missing `using`? (I know it's going to be disposed on context dispose but still prefer to be explicit)